test-data/unit/check-unreachable-code.test

```diff
@@ -915,15 +915,15 @@ def expect_str(x: str) -> str: pass
 
 x: int
 if False:
-    assert False
+    assert False  # E: Statement is unreachable
     reveal_type(x)
```

Comment:
Since `reveal_type(x)` is not a real statement, maybe it should considered as a no-op.